### ISSUE TITLE
import of pandas makes import of torch much slower

### ISSUE THREAD

**Issue by christopherhesse** [pytorch/pytorch#71313]
### 🐛 Describe the bug

`pandas` is imported when you do `import torch`, and this slows down the import of `torch` significantly on my laptop.

Here is the place it is imported: 

https://github.com/pytorch/pytorch/blob/master/torch/utils/data/datapipes/dataframe/dataframe_wrapper.py#L2

If this was replaced with a lazy import (or removed as the comment indicates it will be) then the import times would be significantly better.  On my laptop, uninstall `pandas` changes the import time of `torch` from 1.64s to 1.17s.

### Versions

1.10.1

cc @malfet @seemethere @VitalyFedyunin @ngimel

**Comment by malfet** [comment]
Adding triage review for the same reason as for https://github.com/pytorch/pytorch/issues/71280 (i.e. `module: build` feels like a wrong label here)
